include/ccmath/math/fmanip/ldexp.hpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2024-Present Ian Pike
+ * Copyright (c) 2024-Present ccmath contributors
+ *
+ * This library is provided under the MIT License.
+ * See LICENSE for more information.
+ */
+
+#pragma once
+
+#include <ccmath/internal/support/bits.hpp>
+#include "ccmath/internal/support/floating_point_traits.hpp" //this is included in bits, but i'm going to include it anyways
+
+namespace ccm
+{
+	// this function multiples parameter x by two to the power of paramter powerOf2
+	// x * 2^powerOf2
+	template <typename T, std::enable_if_t<!std::is_integral_v<T>, bool> = true>
+	inline constexpr T ldexp(T x, int32_t powerOf2) noexcept
+	{
+		if (!ccm::isfinite(x)) { return x; }
```

Comment:
Combine this check with the below check that is returning x to simplify the logic.